src/Cryptol/Parser/AST.hs

```diff
@@ -985,4 +990,156 @@ instance NoPos (Prop name) where
       CLocated c _  -> noPos c
       CType t       -> CType (noPos t)
 
+-- --Typeclass for prettying up the AST for exporting to Coq
```

Comment:
* Also, maybe this should be called something else? 
* Add some comments on what's the purpose of these instances.
* I am not quite clear on what's this class is for, but could we get what you need by generalizing `NoPos` rather than having so much duplicated code?